packages/repository/src/mixins/repository.mixin.ts

```diff
@@ -289,7 +290,22 @@ export function RepositoryMixin<T extends MixinTarget<Application>>(
 
         if (operation in ds && typeof ds[operation] === 'function') {
           debug('Migrating dataSource %s', b.key);
-          await ds[operation](options.models);
+          if (!options?.models || options?.models.length === 0) {
+            // if options.models is an empty array, proceed normally
+            await ds[operation](options.models);
+          } else {
+            // if options.models is not empty, check if model is
+            // connected to the datasource
+            const models = [];
+            for (const model of options?.models) {
+              // todo : add "definitions" attribute in Datasource definition
+              // in "loopback-datasource-juggler"
+              if (Object.keys(ds['definitions']).includes(model)) {
+                models.push(model);
+              }
+            }
+            await ds[operation](models);
+          }
```

Comment:
I find the current proposal a bit more complex than I like. How about structuring the code as follows?

```ts
// here, inside migrateSchema()
const models = filterDataSourceModels(options.models, ds);
await ds[operation](models);

// a helper implemented outside of the mixin class
function filterDataSourceModels(models, dataSource) {
  if (!models || models.length === 0) return models;
  const definitions = Object.keys(ds.definitions);
  return models.filter(m => definitions.includes(m));
}
``

Reply:
it's easier to read, true

i'll test it